api/src/State/CampUpdateProcessor.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\State\ProcessorInterface;
+use App\Entity\Camp;
+use App\Entity\CampCollaboration;
+use App\Entity\User;
+use App\State\Util\AbstractPersistProcessor;
+use App\State\Util\PropertyChangeListener;
+use Symfony\Bundle\SecurityBundle\Security;
+
+/**
+ * @template-extends AbstractPersistProcessor<CampCollaboration>
+ */
+class CampUpdateProcessor extends AbstractPersistProcessor {
+    public function __construct(
+        ProcessorInterface $decorated,
+        private Security $security,
+    ) {
+        $sharingChangeListener = PropertyChangeListener::of(
+            extractProperty: fn (Camp $data) => $data->isShared,
+            beforeAction: fn (Camp $data) => $this->onBeforeSharingStatusChange($data),
+        );
+
+        parent::__construct(
+            $decorated,
+            propertyChangeListeners: [
+                $sharingChangeListener,
+            ]
+        );
+    }
+
+    public function onBeforeSharingStatusChange(Camp $data): Camp {
+        if ($data->isShared) {
+            $data->sharedSince = new \DateTime();
```

Comment:
This uses the current UTC time and stores that into the DB. We later interpret the DateTime field without a time zone. So the displayed value is always 1 or 2 hours off... How can I fix this?